PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/storage/azurite/AzuriteSessionDescriptor.kt

```diff
@@ -57,66 +55,79 @@ class AzuriteServiceViewSessionDescriptor(private val project: Project)
-            override fun getIcon(p: Boolean) = superPresentation.getIcon(p)
-            override fun getPresentableText() = superPresentation.presentableText
-        }
+    @Suppress("DialogTitleCapitalization")
+    override fun getPresentation() = PresentationData().apply {
+        locationString = workspace
+        setIcon(CommonIcons.Azurite)
+        addText(RiderAzureBundle.message("service.azurite.name"), SimpleTextAttributes.REGULAR_ATTRIBUTES)
     }
 
-    override fun getContentComponent(): JComponent? {
-        ensureConsoleView()
-        return panel
+    override fun getContentComponent() = BorderLayoutPanel().apply {
+        border = JBUI.Borders.empty()
+        add(consoleView.component)
     }
 
-    private fun ensureConsoleView() {
-        azuriteService.processHandler?.let { activeProcessHandler ->
-
-            if (processHandler != activeProcessHandler) {
-                processHandler?.detachProcess()
-                processHandler = activeProcessHandler
-                workspace = azuriteService.workspace
+    private fun connectToSession(activeProcessHandler: ColoredProcessHandler, activeWorkspace: String) {
+        if (processHandler == activeProcessHandler) return
 
-                consoleView.print(RiderAzureBundle.message("action.azurite.reattach.workspace", workspace.orEmpty()) + "\n", ConsoleViewContentType.LOG_INFO_OUTPUT)
-                consoleView.attachToProcess(activeProcessHandler)
-                consoleView.print(RiderAzureBundle.message("action.azurite.reattach.finished") + "\n", ConsoleViewContentType.LOG_INFO_OUTPUT)
-            }
+        synchronized(myLock) {
+            processHandler?.detachProcess()
+            processHandler = activeProcessHandler
+            workspace = activeWorkspace
         }
 
-        if (processHandler != null && panel.components.isEmpty()) {
-            panel.add(consoleView.component, BorderLayout.CENTER)
-        }
+        consoleView.print(RiderAzureBundle.message("action.azurite.reattach.workspace", activeWorkspace) + "\n", ConsoleViewContentType.LOG_INFO_OUTPUT)
+        consoleView.attachToProcess(activeProcessHandler)
+        consoleView.print(RiderAzureBundle.message("action.azurite.reattach.finished") + "\n", ConsoleViewContentType.LOG_INFO_OUTPUT)
     }
 
-    private fun createEmptyComponent(): JPanel {
-        val panel: JPanel = JBPanelWithEmptyText(BorderLayout())
-                .withEmptyText(RiderAzureBundle.message("service.azurite.not_started"))
-                .withBorder(BorderFactory.createEmptyBorder())
-        panel.isFocusable = true
-        return panel
+    private fun disconnectFromSession() {
+        if (processHandler == null) return
+
+        synchronized(myLock) {
+            processHandler?.detachProcess()
+            processHandler = null
+            workspace = null
+        }
     }
 
     override fun dispose() {
         Disposer.dispose(consoleView)
     }
```

Comment:
_(Minor)_ Note you no longer need this override since you've written `Disposer.register(this, consoleView)`.

Reply:
Thank you, fixed